### PR TITLE
Todo-apps

### PR DESCRIPTION
**Pull request by jchris**

### CODE REVIEW

File: src/blockstore/store-meta.ts [hunk 1]
```diff
@@ -141,11 +149,18 @@ export class MetaStoreImpl extends BaseStoreImpl implements MetaStore {
           await Promise.all(
             dbMetas.map((dbMeta) => this.loader?.taskManager?.handleEvent(dbMeta.eventCid, dbMeta.parents, dbMeta.dbMeta)),
           );
+          this.updateParentsFromDbMetas(dbMetas);
```
Comment (jchris) [inline src/blockstore/store-meta.ts:154]: this was missing

File: src/blockstore/store.ts [hunk 2]
```diff
@@ -230,12 +230,18 @@ export class WALStoreImpl extends BaseStoreImpl implements WALStore {
...
+      );
       this.walState.noLoaderOps.push(dbMeta);
     } else {
       this.walState.operations.push(dbMeta);
     }
     await this.save(this.walState);
-    void this.process();
+    if (!opts.noLoader) {
+      void this.process();
+    }
```
Comment (jchris) [inline src/blockstore/store.ts:243]: this is how it was a while ago

File: src/apply-head-queue.ts [hunk 3]
```diff
@@ -35,8 +35,7 @@ export function applyHeadQueue<T extends DocTypes>(worker: ApplyHeadWorkerFuncti
         const task = queue.shift();
         if (!task) continue;
 
-        // console.time('int_applyHead worker')
-        await worker(task.newHead, task.prevHead, task.updates !== null).catch((e) => {
+        await worker(task.newHead, task.prevHead, task.updates !== undefined).catch((e) => {
```
Comment (jchris) [inline src/apply-head-queue.ts:38]: this makes a difference

File: package.json [hunk 4]
```diff
@@ -93,7 +93,8 @@
     "p-limit": "^6.1.0",
     "prolly-trees": "^1.0.4",
     "randombytes": "^2.1.0",
-    "react": "^18.3.1"
+    "react": "^18.3.1",
+    "uuidv7": "^1.0.2"
```
Comment (jchris) [inline package.json:97]: now we are sorted for newbs

File: src/blockstore/commitor.ts [hunk 5]
```diff
@@ -163,7 +163,7 @@ async function prepareCarFiles(
 ): Promise<{ cid: AnyLink; bytes: Uint8Array }[]> {
   // const theKey = isPublic ? undefined : await this._getKey();
   const carFiles: { cid: AnyLink; bytes: Uint8Array }[] = [];
-  threshold = threshold || 1000 * 1000;
+  threshold = threshold || 128000;
```
Comment (jchris) [inline src/blockstore/commitor.ts]: this low default is for PK testing -- @mabels @mschoch can you help me configure this or the `fragSize` here? https://github.com/fireproof-storage/connect/blob/46f6fcf96cf4a10014affb5c3f4794f768abd017/src/partykit/index.ts#L46

File: src/blockstore/connection-base.ts [hunk 6]
```diff
@@ -76,7 +76,9 @@ export abstract class ConnectionBase implements Connection {
   async connectStorage_X({ loader }: { loader?: Loadable }) {
     if (!loader) throw this.logger.Error().Msg("connectStorage_X: loader is required").AsError();
     this.loader = loader;
-    const dataUrl = this.url.build().defParam("store", "data").URI();
+      // urlObj.searchParams.set("fragSize", "128000");
```
Comment (jchris) [inline src/blockstore/connection-base.ts]: I tried setting fragSize here for testing but it acted strange

File: src/blockstore/fragment-gateway.ts [hunk 7]
```diff
@@ -173,27 +173,28 @@ export class FragmentGateway implements Gateway {
...
+      //       interval *= 2; // Double the interval when data is unchanged
+      //     }
+      //   }
+      //   timeoutId = setTimeout(fetchData, interval);
+      // };
+      // let timeoutId = setTimeout(fetchData, interval);
+
+      // return Result.Ok(() => {
+      //   clearTimeout(timeoutId);
+      // });
```
Comment (jchris) [inline src/blockstore/fragment-gateway.ts]: this shouldn't have been here in the first place

File: src/blockstore/store-meta.ts [hunk 8]
```diff
@@ -11,6 +11,14 @@ import * as rt from "../runtime/index.js";
...
 async function decodeGatewayMetaBytesToDbMeta(sthis: SuperThis, byteHeads: Uint8Array) {
   const crdtEntries = JSON.parse(sthis.txt.decode(byteHeads)) as CRDTEntry[];
+  if (!crdtEntries.length) {
+    sthis.logger.Debug().Str("byteHeads", new TextDecoder().decode(byteHeads)).Msg("No CRDT entries found");
+    return [];
+  }
+  if (!crdtEntries.map) {
+    sthis.logger.Debug().Str("crdtEntries", JSON.stringify(crdtEntries)).Msg("No data in CRDT entries");
+    return [];
+  }
```
Comment (jchris) [inline src/blockstore/store-meta.ts:21]: not sure this is necessary, need to test on empty db etc without it

File: src/blockstore/store-meta.ts [hunk 9]
```diff
@@ -119,6 +127,7 @@ export class MetaStoreImpl extends BaseStoreImpl implements MetaStore {
   readonly storeType = "meta";
   readonly subscribers = new Map<string, LoadHandler[]>();
   parents: CarClockHead = [];
+  remote: boolean;
```
Comment (jchris) [inline src/blockstore/store-meta.ts:130]: this was just for logging

File: src/blockstore/store-meta.ts [hunk 10]
```diff
@@ -180,12 +200,20 @@ export class MetaStoreImpl extends BaseStoreImpl implements MetaStore {
     if (url.isErr()) {
       throw this.logger.Error().Err(url.Err()).Str("branch", branch).Msg("got error from gateway.buildUrl").AsError();
     }
+    console.log("putting event", url.Ok().toString(), event.cid.toString());
+    this.parents = [event.cid];
```
Comment (jchris) [inline src/blockstore/store-meta.ts:200]: moving this before the put made all the difference

File: src/blockstore/store.ts [hunk 11]
```diff
@@ -229,13 +229,18 @@ export class WALStoreImpl extends BaseStoreImpl implements WALStore {
 
   async enqueue(dbMeta: DbMeta, opts: CommitOpts) {
     await this.ready();
-    if (opts.noLoader) {
+    if (opts.compact) {
+      this.walState.operations = [];
+      this.walState.noLoaderOps = [dbMeta];
+    } else if (opts.noLoader) {
```
Comment (jchris) [inline src/blockstore/store.ts:235]: not sure about this...

File: src/database.ts [hunk 12]
```diff
@@ -98,7 +99,7 @@ export class Database<DT extends DocTypes = NonNullable<unknown>> implements Con
     await this.ready();
     this.logger.Debug().Str("id", doc._id).Msg("put");
     const { _id, ...value } = doc;
-    const docId = _id || this.sthis.nextId().str;
+    const docId = _id || uuidv7();
```
Comment (jchris) [inline src/database.ts:102]: easier for new users
